src/main/java/be/woutschoovaerts/mollie/handler/AbstractHandler.java

```diff
@@ -4,177 +4,268 @@
 
-        HttpResponse<String> response = Unirest
-                .delete(url)
-                .headers(configureAllHeaders())
-                .body(body)
-                .asString();
+        try {
+            HttpResponse<String> response = httpClient
+                    .send(httpRequest, HttpResponse.BodyHandlers.ofString());
 
-        validateResponse(response);
-        log.info("Successful response 'DELETE {}'", url);
+            validateResponse(response);
 
-        return response;
+            if (response.statusCode() == 204) {
+                return null;
+            } else {
+                return mapper.readValue(response.body(), clazz);
+            }
+        } catch (IOException | InterruptedException e) {
+            throw new MollieException(e);
+        }
     }
 
-    protected void validateResponse(HttpResponse<String> response) throws IOException, MollieException {
-        if (response.getStatus() < 200 || response.getStatus() > 300) {
+    protected void validateResponse(HttpResponse<String> response) throws MollieException, JsonProcessingException {
+        if (response.statusCode() < 200 || response.statusCode() > 300) {
             log.error("Error response from mollie with status code '{}' and body: {}",
-                    response.getStatus(), response.getBody());
-            throw new MollieException("Error response from mollie", mapper.readValue(response.getBody(),
-                    new TypeReference<Map>() {
+                    response.statusCode(), response.body());
+            throw new MollieException("Error response from mollie", mapper.readValue(response.body(),
+                    new TypeReference<>() {
                     }));
+        } else {
+            log.debug("Successful response '{} {}'", response.request().method(), response.uri());
         }
     }
 
+    protected HttpResponse<String> validateASyncResponse(HttpResponse<String> response) {
+        if (response.statusCode() < 200 || response.statusCode() > 300) {
+            log.error("Error response from mollie with status code '{}' and body: {}",
+                    response.statusCode(), response.body());
+            try {
+                throw new CompletionException("Error response from mollie", mapper.readValue(response.body(),
+                        new TypeReference<>() {
+                        }));
+            } catch (JsonProcessingException e) {
+                throw new CompletionException("Error parsing response from mollie", e);
+            }
+        }
+        return response;
+    }
+
     private Map<String, String> configureAllHeaders() {
         Map<String, String> map = new HashMap<>();
 
         map.put("Content-Type", "application/json");
         map.put("Authorization", "Bearer " + config.getBearerToken());
+        config.getProxy().ifPresent(proxy -> {
+            if (StringUtils.isNotEmpty(proxy.getUsername()) && StringUtils.isNotEmpty(proxy.getPassword())) {
```

Comment:
Maybe put the if statement into 2 filter() calls

Will make it a bit more readable I think